src/Hadrian/Utilities.hs

```diff
@@ -453,3 +460,21 @@ renderUnicorn ls =
     ponyPadding = "                                            "
     boxLines :: [String]
     boxLines = ["", "", ""] ++ (lines . renderBox $ ls)
+
+-- | These arguments are used by the `test` target.
+data TestArgs = TestArgs
```

Comment:
I'm not sure why `TestArgs` is defined here, let's move it to `CmdLineArgs`.

Everything in `src/Hadrian` is supposed to be generally useful for build systems, i.e. not GHC-specific, whereas flags passed to the GHC's testsuite are very much GHC-specific.